src/PhpPact/Consumer/Model/Pact.php

```diff
@@ -0,0 +1,189 @@
+    }
+
+    public function createMockServer(): void
+    {
+        $port = $this->ffi->pactffi_create_mock_server_for_transport(
+            $this->id,
+            $this->config->getHost(),
+            $this->config->getPort(),
+            $this->config->isSecure() ? 'https' : 'http',
+            null
+        );
+
+        if ($port < 0) {
+            $message = match ($port) {
+                -1 => 'An invalid handle was received. Handles should be created with `pactffi_new_pact`',
+                -2 => 'Transport_config is not valid JSON',
+                -3 => 'The mock server could not be started',
+                -4 => 'The method panicked',
+                -5 => 'The address is not valid',
+            };
+            throw new MockServerNotStartedException($message);
+        }
+        $this->config->setPort($port);
+    }
+
+    public function verifyInteractions(): bool
+    {
+        $result = $this->ffi->pactffi_mock_server_matched($this->config->getPort());
+
+        if ($result) {
+            $error = $this->ffi->pactffi_write_pact_file(
+                $this->config->getPort(),
+                $this->config->getPactDir(),
+                $this->config->getPactFileWriteMode() === PactConfigInterface::MODE_OVERWRITE
+            );
+            if ($error) {
+                $message = match ($error) {
+                    1 => 'A general panic was caught',
+                    2 => 'The pact file was not able to be written',
+                    3 => 'A mock server with the provided port was not found',
+                };
+                throw new PactFileNotWroteException($message);
+            }
+        }
+
+        $this->ffi->pactffi_cleanup_mock_server($this->config->getPort());
+        $this->ffi->pactffi_free_pact_handle($this->id);
+
+        return $result;
+    }
+
+    public function registerInteraction(Interaction $interaction): bool
+    {
+        $this
+            ->newInteraction($interaction)
+            ->given($interaction)
+            ->uponReceiving($interaction)
+            ->with($interaction)
+            ->willRespondWith($interaction);
+
+        return true;
+    }
+
+    private function initWithLogLevel(): self
+    {
+        $logLevel = $this->config->getLogLevel();
+        if ($logLevel) {
+            $this->ffi->pactffi_init_with_log_level($logLevel);
+        }
+
+        return $this;
+    }
+
+    private function newPact(): self
+    {
+        $this->id = $this->ffi->pactffi_new_pact($this->config->getConsumer(), $this->config->getProvider());
+
+        return $this;
+    }
+
+    private function withSpecification(): self
+    {
+        $supportedVersions = [
+            '1.0.0' => $this->ffi->PactSpecification_V1,
+            '1.1.0' => $this->ffi->PactSpecification_V1_1,
+            '2.0.0' => $this->ffi->PactSpecification_V2,
+            '3.0.0' => $this->ffi->PactSpecification_V3,
+            '4.0.0' => $this->ffi->PactSpecification_V4,
+        ];
```

Comment:
Hard coded. Do you have any better idea?

Related to https://github.com/pact-foundation/pact-php/pull/291

Reply:
Resolve to make code reviewing easier. Feel free to reopen or add comment.

Reply:
Updated. Use semver to compare version.